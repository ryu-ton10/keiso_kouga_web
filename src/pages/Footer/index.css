.footer {
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;
}

/* ========= プライバシーポリシー表示用リンク =========== */
.privacy_policy {
  padding: 20px 0 20px 0;
}

.privacy_policy p {
  margin-bottom: 0;
  color: #808080;
  border-bottom: solid 2px #808080;
}

.privacy_policy p:hover {
  color: #a9a9a9;
}
