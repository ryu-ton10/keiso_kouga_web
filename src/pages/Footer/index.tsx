import React, { useState } from 'react';
import PrivacyPolicy from 'components/PrivacyPolicy';
import './index.css';

type Props = {
  privacy_policy: string;
  twitter_url: string;
};

function Footer(props: Props) {
  
  const { privacy_policy, twitter_url } = props;
  const [isShowModal, setIsShowModal] = useState(false);
  
  return (
    <div className="footer" data-testid="footer">
      <div className="privacy_policy">
        <p onClick={ () => setIsShowModal(true) }>プライバシーポリシー</p>
      </div>
      {isShowModal &&
        <PrivacyPolicy scentence={privacy_policy} hook={setIsShowModal} />
      }
      <p>©︎2022    <a href={twitter_url} rel="noopener noreferrer" target="_blank">@meitonto37</a></p>
    </div>
  );
};

export default Footer;
