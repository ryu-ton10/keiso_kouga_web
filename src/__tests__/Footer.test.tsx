import renderer from 'react-test-renderer';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import Footer from 'pages/Footer';

let container = null;
let root = null;
const dummy_privacy_policy = 'hogehoge';
const dummy_twitter_url = 'fugafuga';


beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  root = createRoot(container);
});

afterEach(() => {
  root.unmount(container);
  container.remove();
  container = null;
});

/* ---------------- snapshot test -------------------- */
it('コンポーネントが描画されること', () => {
  const component = renderer.create(
    <Footer
      privacy_policy={dummy_privacy_policy}
      twitter_url={dummy_twitter_url}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})


/* ---------------- act test -------------------- */
it('ボタンをクリックするとプライバシーポリシーが表示されること', () => {
  act(() => {
    root.render(
      <Footer
        privacy_policy={dummy_privacy_policy}
        twitter_url={dummy_twitter_url}
      />
    );
  });

  const button = container.querySelector("[data-test-id=privacy_policy_button]");

  // プライバシーポリシーボタンであることの確認
  expect(button.textContent).toBe('プライバシーポリシー');

  // ボタンのクリック
  act(() => {
    button.dispatchEvent(new MouseEvent('click', {bubbles: true}));
  });

  expect(container.querySelector(".modal")).toBeTruthy();
})
