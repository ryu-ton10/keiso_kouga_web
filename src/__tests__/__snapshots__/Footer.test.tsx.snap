// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`コンポーネントが描画されること 1`] = `
<div
  className="footer"
  data-testid="footer"
>
  <div
    className="privacy_policy"
  >
    <p
      onClick={[Function]}
    >
      プライバシーポリシー
    </p>
  </div>
  <p>
    ©︎2022    
    <a
      href="fugafuga"
      rel="noopener noreferrer"
      target="_blank"
    >
      @meitonto37
    </a>
  </p>
</div>
`;
